packages/fetchai/connections/p2p_noise/aea/api.go

```diff
@@ -0,0 +1,395 @@
+  data, err := proto.Marshal(envelope)
+  if err != nil {
+    fmt.Println("[aea-api  ][error] while serializing envelope:", envelope, ":", err)
+    return err
+  }
+  return write(pipe, data)
+}
+
+func read_envelope(pipe *os.File) (*Envelope, error) {
+  envelope := &Envelope{}
+  data, err := read(pipe)
+  if err != nil {
+    fmt.Println("[aea-api  ][error] while receiving data:", err)
+    return envelope, err
+  }
+  err = proto.Unmarshal(data, envelope)
+  return envelope, err
+}
+
+/*
+  
+  Sandbox
+
+*/
+
+func setup_aea_sandbox() (string, string, string, net.IP, uint16, error) {
+  // setup id
+  id := ""
+  // setup uri
+  host := net.ParseIP("127.0.0.1")
+  port := uint16(5000 + rand.Intn(10000))
+  // setup pipes
+  ROOT_PATH := "/tmp/aea_sandbox_" + strconv.FormatInt(time.Now().Unix(), 10)
+  msgin_path  := ROOT_PATH + ".in"
+  msgout_path := ROOT_PATH + ".out"
+  // create pipes
+  if _, err := os.Stat(msgin_path); ! os.IsNotExist(err) {
+    os.Remove(msgin_path)
+  }
+  if _, err := os.Stat(msgout_path); ! os.IsNotExist(err) {
+    os.Remove(msgout_path)
+  }
+  erri := syscall.Mkfifo(msgin_path, 0666)
+  erro := syscall.Mkfifo(msgout_path, 0666)
+  if erri != nil || erro != nil {
+    fmt.Println("[aea-api  ][error][sandbox] setting up pipes:", erri, erro)
+    if erri != nil {
+      return  "", "", "", nil, 0, erri
+    }
+    return "", "", "", nil, 0, erro 
+  }
+  go run_aea_sandbox(msgin_path, msgout_path)
+  return msgin_path, msgout_path, id, host, port, nil
+}
+
+
+func run_aea_sandbox(msgin_path string, msgout_path string) error {
+  // open pipe
+  msgout, erro := os.OpenFile(msgout_path, os.O_RDONLY, os.ModeNamedPipe)
+  msgin,  erri := os.OpenFile(msgin_path, os.O_WRONLY, os.ModeNamedPipe)
+  if erri != nil || erro != nil {
+    fmt.Println("[aea-api  ][error][sandbox] error while opening pipes:", erri, erro)
+    if erri != nil {
+      return erri
+    } else {
+      return erro
+    }
+  }
+  
+  // consume envelopes
+  go func() {
+    for {
+      envel, err := read_envelope(msgout)
+      if err != nil {
+        fmt.Println("[aea-api  ][error][sandbox] stopped receiving envelopes:", err)
+        return
+      }
+      fmt.Println("[aea-api  ][error][sandbox] consumed envelope", envel)
+    }
+  }()
+
+  // produce envelopes
+  go func() {
+    i := 1
+    for {
+      time.Sleep(time.Duration((rand.Intn(5000)+3000))*time.Millisecond)
+      envel := &Envelope{"aea-sandbox", "golang", "fetchai/default:0.1.0", []byte("\x08\x01*\x07\n\x05Message from sandbox "+strconv.Itoa(i)), ""}
+      err := write_envelope(msgin, envel)
+      if err != nil {
+        fmt.Println("[aea-api  ][error][sandbox] stopped producing envelopes:", err)
+        return
+      }
+      i += 1
+    }
+  }()
+
+  return nil
+}
+
+
+/*
+  
+  Protobuf generated Envelope - Edited
+
+*/
+
+// Code generated by protoc-gen-go. DO NOT EDIT.
+// source: pocs/p2p_noise_pipe/envelope.proto
```

Comment:
not sure my comment makes sense. I believe `api.go` has been written by you. However, the code below this comment seems to be autogenerated. I wonder if it would be possible to put this in a different file and then add it into the compilation, so to make updates of the package easier (instead of copying & pasting the generated code into this).

Reply:
This part is generated by protoc and *edited* by me to be able to use the struct as a message type for noise.